src/history/repeat_commands/repeat_commands_specific.ts

```diff
@@ -0,0 +1,156 @@
+  InsertCellCommand,
+  PasteCommand,
+  RemoveColumnsRowsCommand,
+  RepeatPasteCommand,
+  ResizeColumnsRowsCommand,
+  SortCommand,
+} from "./../../types/commands";
+import { transformSheetDependantCommand } from "./repeat_commands_generic";
+
+const uuidGenerator = new UuidGenerator();
+
+export function repeatCreateChartCommand(
+  getters: Getters,
+  cmd: CreateChartCommand
+): CreateChartCommand {
+  return {
+    ...transformSheetDependantCommand(getters, cmd),
+    id: uuidGenerator.uuidv4(),
+  };
+}
+
+export function repeatCreateImageCommand(
+  getters: Getters,
+  cmd: CreateImageOverCommand
+): CreateImageOverCommand {
+  return {
+    ...transformSheetDependantCommand(getters, cmd),
+    figureId: uuidGenerator.uuidv4(),
+  };
+}
+
+export function repeatCreateFigureCommand(
+  getters: Getters,
+  cmd: CreateFigureCommand
+): CreateFigureCommand {
+  const newCmd = transformSheetDependantCommand(getters, cmd);
+  newCmd.figure.id = uuidGenerator.uuidv4();
+  return newCmd;
+}
+
+export function repeatCreateSheetCommand(
+  getters: Getters,
+  cmd: CreateSheetCommand
+): CreateSheetCommand {
+  const newCmd = deepCopy(cmd);
+  newCmd.sheetId = uuidGenerator.uuidv4();
+
+  const sheetName = cmd.name || getters.getSheet(getters.getActiveSheetId()).name;
+  // Extract the prefix of the sheet name (everything before the number at the end of the name)
+  const namePrefix = sheetName.match(/(.+?)\d*$/)?.[1] || sheetName;
+
+  newCmd.name = getters.getNextSheetName(namePrefix);
+
+  return newCmd;
+}
+
+export function repeatDuplicateSheetCommand(
+  getters: Getters,
+  cmd: DuplicateSheetCommand
+): DuplicateSheetCommand | undefined {
+  if (!getters.tryGetSheet(cmd.sheetId)) return undefined;
+  return {
+    ...deepCopy(cmd),
+    sheetIdTo: uuidGenerator.uuidv4(),
+  };
+}
+
+export function repeatAddColumnsRowsCommand(
+  getters: Getters,
+  cmd: AddColumnsRowsCommand
+): AddColumnsRowsCommand {
+  const currentSelection = getters.getSelectedZones();
+  return {
+    ...transformSheetDependantCommand(getters, cmd),
+    base: cmd.dimension === "COL" ? currentSelection[0].left : currentSelection[0].top,
+  };
+}
+
+export function repeatHeaderElementCommand<
+  T extends RemoveColumnsRowsCommand | HideColumnsRowsCommand | ResizeColumnsRowsCommand
+>(getters: Getters, cmd: T): T {
+  const currentSelection = getters.getSelectedZones();
+  return {
+    ...transformSheetDependantCommand(getters, cmd),
+    elements: [cmd.dimension === "COL" ? currentSelection[0].left : currentSelection[0].top],
+  };
+}
+
+export function repeatInsertOrDeleteCellCommand<T extends InsertCellCommand | DeleteCellCommand>(
+  getters: Getters,
+  cmd: T,
+  childCommands: readonly CoreCommand[]
+): T {
+  const currentSelection = getters.getSelectedZones();
+  return {
+    ...deepCopy(cmd),
+    zone: currentSelection[0],
+  };
+}
+
+export function repeatAutoresizeCommand<T extends AutoresizeColumnsCommand | AutoresizeRowsCommand>(
```

Comment:
```suggestion
export function repeatAutoResizeCommand<T extends AutoresizeColumnsCommand | AutoresizeRowsCommand>(
```
Let's not repeat mistakes from the past